### ISSUE TITLE
SPI doesn't work as expected on MSP432 RED

### ISSUE THREAD

**Issue by rei-vilo** [energia/msp432r-core#9]
SPI doesn't work as expected on MSP432 RED. 

SPI modes seem to be mixed on MSP432R in silicon. 

+ `Mode0` and `Mode1` are inverted. 
+ `Mode2`and `Mode3` are inverted.

For more information, please refer to [MSP432P401R Device Erratasheet (Rev. O)](http://www.ti.com/lit/er/slaz610o/slaz610o.pdf).

Different options:

+ Try building against against **LaunchPad w/ MSP432 EMT (48 MHz)** and not **RED LaunchPad w/ MSP432 EMT (48 MHz)**

<img width="330" alt="capture 2018-05-05 a 10 53 41" src="https://user-images.githubusercontent.com/951776/39661566-d8f0feae-5053-11e8-9b76-52a1ab13a682.png">

+ Edit file `~/Library/Energia15/packages/energia/hardware/msp432/3.8.0/libraries/EduBPMKII_Screen/Screen_HX8353E.cpp` and replace line 92

```
    SPI.setDataMode(SPI_MODE0);
```

by

```
#if defined(ENERGIA_ARCH_MSP432R)
#warning Horrible patch for hardware bug
    SPI.setDataMode(SPI_MODE1);
#else
    SPI.setDataMode(SPI_MODE0);
#endif
```

<img width="585" alt="capture 2018-05-05 a 10 55 56" src="https://user-images.githubusercontent.com/951776/39661563-d5755130-5053-11e8-9c64-841a5a3267ca.png">

**Comment by gerz122** [comment]
I tried this and it worked for me. 

I used the RED LaunchPad w/ MSP432 EMT (48 MHz) to build the project.

I find the Screen_HX8353E.cpp file in a temporary folder created by windows. When compiling the project, it is possible to see the symbols path. I found the file starting from this root path.

![image](https://user-images.githubusercontent.com/39070435/39729625-c106bcf6-5211-11e8-8ab9-1553aeebbc83.png)


This works as a software fix that allows using the boosterpack mkII LCD screen HX8353E with the red launchpad msp432p401r.
